app/parse.py

```diff
@@ -15,9 +29,104 @@ class Product:
     num_of_reviews: int
 
 
+def scrape_single_product(product: WebElement) -> Product:
+    title = product.find_element(By.CLASS_NAME, "title").get_attribute("title")
+    price, _, description, num_of_reviews = product.text.split("\n")
```

Comment:
The line `price, _, description, num_of_reviews = product.text.split("\n")` assumes a specific format of the product text. If the format changes or if there are missing elements, this line will raise a `ValueError`. Consider adding error handling or checking the length of the split result to ensure robustness.